### PR TITLE
2004 freeze lists

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Freezes lists in configuration object  #2004

## Testing Checklist

* [ ] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by running UT's
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/common/configuration/default_agent_configuration.py [hunk 1]
```diff
@@ -64,37 +64,38 @@
...
 TCP_SCAN_CONFIGURATION = TCPScanConfiguration(timeout=3.0, ports=TCP_PORTS)
 ICMP_CONFIGURATION = ICMPScanConfiguration(timeout=1.0)
-HTTP_PORTS = [80, 443, 7001, 8008, 8080, 8983, 9200, 9600]
-FINGERPRINTERS = [
+HTTP_PORTS = (80, 443, 7001, 8008, 8080, 8983, 9200, 9600)
+FINGERPRINTERS = (
     PluginConfiguration("elastic", {}),
-    PluginConfiguration("http", {"http_ports": HTTP_PORTS}),
+    # Plugin configuration option contents are not converted to tuples
+    PluginConfiguration("http", {"http_ports": list(HTTP_PORTS)}),
```
Comment (VakarisZ) [inline monkey/common/configuration/default_agent_configuration.py:75]: We can't know the contents of plugin options so we can't freeze it easily. It's also a dict and there's no frozen dict in python.

Comment (mssalvatore) [reply monkey/common/configuration/default_agent_configuration.py:75]: We can wrap `dicts`/`mappings` in `MappingProxyType`. https://docs.python.org/3/library/types.html#types.MappingProxyType

File: monkey/common/utils/code_utils.py [hunk 2]
```diff
@@ -48,3 +48,10 @@ def del_key(mapping: MutableMapping[T, Any], key: T):
         del mapping[key]
     except KeyError:
         pass
+
+
+def freeze_lists_in_dict(mapping: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+    for key, value in mapping.items():
+        if type(value) == list:
```
Comment (mssalvatore) [inline monkey/common/utils/code_utils.py:55]:
```suggestion
        if type(value) == Sequence:
```
or
```suggestion
        if isinstance(value, Sequence)
```

File: monkey/common/configuration/default_agent_configuration.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/common/configuration/default_agent_configuration.py:75]: ^ We can probably skip it for this for now.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**